runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java

```diff
@@ -0,0 +1,182 @@
+	tags = {"thieving", "stealing", "artefact", "artefacts", "captain", "khaled"},
+	enabledByDefault = false
+)
+public class StealingArtefactsPlugin extends Plugin
+{
+	@Inject
+	private Client client;
+
+	@Inject
+	private ClientThread clientThread;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private StealingArtefactsOverlay overlay;
+
+	@Getter(AccessLevel.PACKAGE)
+	private StealingArtefactState stealingArtefactState;
+
+	@Getter(AccessLevel.PACKAGE)
+	private boolean isInPortPiscariliusRegion;
+
+	private static final ImmutableSet<Integer> PORT_PISCARILIUS_REGIONS = ImmutableSet.of(6970, 7226);
+
+	private static final ImmutableSet<Integer> ARTEFACT_ITEM_IDS = ImmutableSet.of(ItemID.STOLEN_PENDANT,
+		ItemID.STOLEN_GARNET_RING, ItemID.STOLEN_CIRCLET, ItemID.STOLEN_FAMILY_HEIRLOOM, ItemID.STOLEN_JEWELRY_BOX);
+
+	private NPC captainKhaled = null;
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		overlayManager.add(overlay);
+
+		if (client.getGameState() == GameState.LOGGED_IN)
+		{
+			clientThread.invoke(() ->
+				stealingArtefactState = StealingArtefactState.getStealingArtefactState(client.getVar(Varbits.STEALING_ARTEFACT_STATE)));
+		}
+		else
+		{
+			stealingArtefactState = null;
+		}
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		client.clearHintArrow();
+		overlayManager.remove(overlay);
+	}
+
+	private boolean hasArtefact()
+	{
+		ItemContainer inventory = client.getItemContainer(InventoryID.INVENTORY);
+
+		if (inventory != null)
+		{
+			final Stream<Item> items = Arrays.stream(inventory.getItems());
+			return items.anyMatch(item -> ARTEFACT_ITEM_IDS.contains(item.getId()));
+		}
+		return false;
+	}
+
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged event)
+	{
+		int stealingArtefactStateVarbit = client.getVar(Varbits.STEALING_ARTEFACT_STATE);
+		StealingArtefactState state = StealingArtefactState.getStealingArtefactState(stealingArtefactStateVarbit);
+
+		if (state != StealingArtefactState.DELIVER_ARTEFACT)
+		{
+			stealingArtefactState = state;
+			if (state != StealingArtefactState.NO_TASK && state != StealingArtefactState.FAILURE)
+			{
+				client.setHintArrow(stealingArtefactState.getWorldPoint());
+			}
+			else
+			{
+				client.clearHintArrow();
+			}
+		}
+		else if (captainKhaled != null)
+		{
+			client.setHintArrow(captainKhaled);
+		}
+		else
+		{
+			client.clearHintArrow();
+		}
+	}
+
+	@Subscribe
+	public void onItemContainerChanged(final ItemContainerChanged event)
+	{
+		if (event.getItemContainer() == client.getItemContainer(InventoryID.INVENTORY))
+		{
+			if (hasArtefact())
+			{
+				stealingArtefactState = StealingArtefactState.DELIVER_ARTEFACT;
+			}
+			else if (stealingArtefactState == StealingArtefactState.DELIVER_ARTEFACT)
+			{
+				stealingArtefactState = StealingArtefactState.FAILURE;
+			}
+		}
+	}
+
+	@Subscribe
+	public void onNpcSpawned(NpcSpawned event)
```

Comment:
`NpcDespawned` also needs to be handled. Also, I believe NPCs are GC'd on game state change without firing despawn events? That must also be handled.

Reply:
Alright I've just added `onNpcDespawn()` handling. What is GC in this context?

Reply:
Garbage collected. You need to reset any stored NPCs on game state change:

```java
@Subscribe
public void onGameStateChanged(GameStateChanged event)
{
	if (event.getGameState() != LOGGED_IN)
	{
		captainKhaled = null;
	}
}
```